src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/AvailableLocationsService.kt

```diff
@@ -51,19 +58,16 @@ class AvailableLocationsService(
         while (meetingStartTime.isBefore(endOfDay) && this.size < maxSlots) {
           if (!bookedLocations.isBooked(location, meetingStartTime, meetingEndTime) &&
             request.fallsWithinSlotTime(meetingStartTime) &&
-            location.allowsByAnyRuleOrSchedule(request)
+            location.allowsByAnyRuleOrSchedule(request, meetingStartTime)
           ) {
-            // TODO need to check against the room decoration/schedules
             add(
               AvailableLocation(
-                // TODO is it possible for the description to be null?
-                name = location.description ?: "UNKNOWN",
+                name = location.description ?: location.key,
```

Comment:
It feels like putting the key here should suffice on the off chance there is no description for a location.